### PR TITLE
Migrate to DeserializeGuard

### LINKED ISSUE #237: STS restarter should only watch secret/cm metadata
It's a lot of completely unused data to store, and especially *sensitive* data that makes c-o into a particularly juicy target to attack (though it's still suboptimal that it needs to have access to it in the first place...).

### PR DESCRIPTION
**Pull request by nightkr**
# Description

Some cases have been migrated to PartialObjectMeta instead where only the metadata is used anyway.

Part of stackabletech/issues#211. Fixes #237.

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [x] Helm chart can be installed and deployed operator works
- [x] Integration tests passed (for non trivial changes)
- [x] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

**Review by razvan — APPROVED** [review]
lgtm!

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,10 +16,13 @@ All notable changes to this project will be documented in this file.
 ### Fixed
 
 - BREAKING: The fields `connection` and `host` on `S3Connection` as well as `bucketName` on `S3Bucket`are now mandatory. Previously operators errored out in case these fields where missing ([#283]).
+- Failing to parse one `ZookeeperCluster`/`ZookeeperZnode` should no longer cause the whole operator to stop functioning ([#293]).
+- The StatefulSet restarter service now only retrieves metadata for ConfigMaps and Secrets, rather than full objects ([#293]).
```
Comment (sbernauer) [inline CHANGELOG.md:20]: Did not really look into the code, but does this also reduce the memory usage?

Comment (nightkr) [reply CHANGELOG.md:20]: It should, yes.

Comment (sbernauer) [reply CHANGELOG.md:20]: Awesome!

**Review by razvan — APPROVED** [review]

**Pull request merged**